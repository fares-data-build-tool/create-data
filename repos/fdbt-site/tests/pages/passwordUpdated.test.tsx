import * as React from 'react';
import { shallow } from 'enzyme';
import PasswordUpdated, { getServerSideProps } from '../../src/pages/passwordUpdated';
import { getMockContext } from '../testData/mockData';

describe('passwordUpdated', () => {
    it('should render correctly', () => {
        const redirectTo = '/';
        const tree = shallow(<PasswordUpdated redirectTo={redirectTo} />);
        expect(tree).toMatchSnapshot();
    });

    describe('getServerSideProps', () => {
        it.each([
            ['/login', '/resetPassword'],
            ['/account', '/changePassword'],
        ])("should return redirectTo as '%s' when the user has come from the '%s' page", (redirectTo, redirectFrom) => {
            const mockUserCookieValue = { redirectFrom };
            const ctx = getMockContext({ cookies: { userCookieValue: mockUserCookieValue } });
            const props = getServerSideProps(ctx);
            expect(props).toEqual({ props: { redirectTo } });
        });

        it("should return redirectTo as '/' when the USER_COOKIE is missing", () => {
            const ctx = getMockContext();
            const props = getServerSideProps(ctx);
            expect(props).toEqual({ props: { redirectTo: '/account' } });
        });
    });
});
