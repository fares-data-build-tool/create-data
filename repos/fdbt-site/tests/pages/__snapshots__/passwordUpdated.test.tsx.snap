// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`passwordUpdated should render correctly 1`] = `
<Component
  description="Password Updated page of the Fares Data Build Tool"
  title="Password Updated - Fares Data Build Tool"
>
  <h1
    className="govuk-heading-l"
  >
    Password Updated
  </h1>
  <p
    className="govuk-body-l"
  >
    Your password has been updated successfully
  </p>
  <a
    className="govuk-button"
    data-module="govuk-button"
    draggable="false"
    href="/"
    id="continue-button"
    role="button"
  >
    Continue
  </a>
</Component>
`;
